src/Microsoft.AspNetCore.Mvc.Analyzers/CodeAnalysisExtensions.cs

```diff
@@ -123,6 +123,24 @@ public static bool IsAssignableFrom(this ITypeSymbol source, ITypeSymbol target)
             return false;
         }
 
+        public static bool IsCollection(this ITypeSymbol source)
+        {
+            if (source.Kind == SymbolKind.ArrayType)
+            {
+                return true;
+            }
+
+            foreach (var @interface in source.AllInterfaces)
+            {
+                if (@interface.MetadataName == "IEnumerable")
```

Comment:
I'd recommend checking the actual type `@interface == symbolCache.IEnumerable`. You can get to `IEnumerable` using 
`Compilation.GetSpecialType(SpecialType.System_Collections_IEnumerable)`

Reply:
Sure